swaggen/src/main/java/generator/PathGenerator.java

```diff
@@ -54,9 +65,86 @@
 	 * @param annotation the annotation
 	 * @return the path
 	 */
-	private static Map<RequestMethod, Endpoint> generatePathFromAnnotation(SwaggerGen annotation) {
-		Map<RequestMethod, Endpoint> endpointMap = new HashMap<>();
-		endpointMap.put(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation));
-		return endpointMap;
+	private static SwaggerEndpoint generatePathFromAnnotation(SwaggerGen annotation, SwaggerGenClass klassAnnotation) {
+		SwaggerEndpoint endpoint = new SwaggerEndpoint();
+		endpoint.addEndpoint(RequestMethod.valueOf(annotation.method()), EndpointFactory.createEndpoint(annotation, klassAnnotation));
+		
+		return endpoint;
+	}
+
+	/**
+	 * Checks if URL is already in the existing map, and if it is, appends the swagger
+	 * endpoints to the URL
+	 * Case: multiple classes with the same endpoint
+	 * @param existingPaths Map of existing Swagger Endpoints
+	 * @param classPaths Map of new Swagger Endpoints in the class
+	 */
+	private static Map<PathInfo, SwaggerEndpoint> checkClassEndpoints(Map<PathInfo, SwaggerEndpoint> existingPaths, Map<PathInfo, SwaggerEndpoint> classPaths) {
+		for (PathInfo existingPath: existingPaths.keySet()) {
+			for (PathInfo pathKey : classPaths.keySet()) {
+				if ((existingPath.getBasePath().equals(pathKey.getBasePath())) && 
+					(existingPath.getURI().equals(pathKey.getURI()))) {
+					existingPaths.get(existingPath).addEndpoint(classPaths.get(pathKey));
+					classPaths.remove(pathKey);
+				}
+			}
+		}
+		existingPaths.putAll(classPaths);
+		return existingPaths;
+	}
+
+	/**
+	 * Checks if there are any other request methods within a class
+	 * and puts it in the swagger endpoints map
+	 * Case: multiple request methods within the same class
+	 * @param annotation SwaggerGen annotation
+	 * @param existingPaths Map of existing Swagger Endpoints
+	 */
+	private static void checkRequestMethods(SwaggerGen annotation, Map<PathInfo, SwaggerEndpoint> existingPaths, SwaggerGenClass klassAnnotation) {
+		if (annotation == null) {
+			throw new IllegalArgumentException("annotation cannot be null");
+		}
+		PathInfo pathInfo = addPathInfo(annotation, klassAnnotation);
+
+		if (existingPaths.isEmpty()) {
+			existingPaths.put(pathInfo, generatePathFromAnnotation(annotation, klassAnnotation));
+		}
+		for (PathInfo existingPath: existingPaths.keySet()) {
+			if ((existingPath.getBasePath().equals(pathInfo.getBasePath())) && 
+			(existingPath.getURI().equals(pathInfo.getURI()))) {

```

Comment:
4 spaces (or 1 tab) for a nested statement. 8 spaces (or 2 tabs) for a continued statement on a new line. This way it is easy to distinguish a continued `if` statement from the first statement in the block.